### PR TITLE
Feature/ms wells - part 2: Solving, straightforward option

### PR DESCRIPTION
**Pull request by lisajulia**
This PR is based on https://github.com/OPM/opm-simulators/pull/5680.

This PR enables simulations with MSWell distributed across several processes. The PR https://github.com/OPM/opm-simulators/pull/5680 covered the assembly of the system, this PR covers the straightforward option of solving.
This PR adds communication when multiplying x_well with the matrix B of the Schur complement as explained on slide 13 of [24-10-21-MSWells.pdf](https://github.com/user-attachments/files/17869352/24-10-21-MSWells.pdf).
This approach is the easiest way to solve, using preconditioners for the well system is to be implemented in the future.

I've added two tests comparing the SMRY files of a parallel and a sequential simulation with distributet MSWells. These tests fail in the PR https://github.com/OPM/opm-simulators/pull/5747 because they do not run through, for this PR they pass.

### CODE REVIEW

**Comment by lisajulia** [comment]
jenkins build this failure_report please

**Comment by lisajulia** [comment]
jenkins build this failure_report please

**Comment by lisajulia** [comment]
jenkins build this failure_report please

**Comment by lisajulia** [comment]
jenkins build this failure_report please

**Comment by lisajulia** [comment]
jenkins build this opm-tests=1250 failure_report please

**Comment by lisajulia** [comment]
jenkins build this opm-tests=1250 failure_report please

**Comment by lisajulia** [comment]
jenkins build this opm-tests=1250 please

**Comment by lisajulia** [comment]
jenkins build this opm-tests=1250 please

**Comment by lisajulia** [comment]
jenkins build this opm-tests=1250 please

**Comment by lisajulia** [comment]
jenkins build this opm-tests=1250 please

**Comment by lisajulia** [comment]
jenkins build this opm-tests=1250 please

**Comment by lisajulia** [comment]
jenkins build this failure_report opm-tests=1250 please

File: opm/simulators/wells/MultisegmentWellEquations.cpp [hunk 1]
```diff
@@ -193,24 +202,39 @@ template<class Scalar, int numWellEq, int numEq>
...
+    // the single process to complete the computation.
     return mswellhelpers::applyUMFPack(*duneDSolver_, rhs);
 }
 
 template<class Scalar, int numWellEq, int numEq>
 void MultisegmentWellEquations<Scalar,numWellEq,numEq>::
 recoverSolutionWell(const BVector& x, BVectorWell& xw) const
 {
-    BVectorWell resWell = resWell_;
+    BVectorWell Bx(duneB_.N());
```
Comment (blattms) [inline opm/simulators/wells/MultisegmentWellEquations.cpp:225]: I would recommend to use the old way, when the well is on one process here.

Maybe we can also factor this out similar as for [standard wells](https://github.com/OPM/opm-simulators/blob/53cf22d516f3569bdf3019f4cbddd8158b7e87f2/opm/simulators/wells/WellHelpers.cpp#L106-L126), or refactor that code to reuse here.

I am a bit late, sorry.

**Comment by lisajulia** [comment]
jenkins build this failure_report opm-tests=1250 please

**Comment by lisajulia** [comment]
jenkins build this failure_report opm-tests=1250 please

**Comment by lisajulia** [comment]
jenkins build this opm-tests=1250 please

**Comment by lisajulia** [comment]
jenkins build this opm-tests=1250 please

**Review by blattms — APPROVED** [review]

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
jenkins build this opm-tests=1250 serial please

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
jenkins build this please

**Comment by akva2** [comment]
there's an up-to 5 min window between merge in opm-tests and update on jenkins. we use a shared copy to reduce network traffic and the update polls at 5 min intervals. the printout is there precisely so we can identify such mishaps.

**Comment by blattms** [comment]
Strange CMake failure now:
```
-- The CUDA compiler identification is unknown
CMake Error at /usr/share/cmake-3.25/Modules/CMakeDetermineCUDACompiler.cmake:603 (message):
  Failed to detect a default CUDA architecture.

  

  Compiler output:

Call Stack (most recent call first):
  CMakeLists.txt:200 (enable_language)


-- Configuring incomplete, errors occurred!
See also "/var/lib/jenkins/workspace/opm-simulators-PR-builder/mpi-shared/build-opm-simulators/CMakeFiles/CMakeOutput.log".
See also "/var/lib/jenkins/workspace/opm-simulators-PR-builder/mpi-shared/build-opm-simulators/CMakeFiles/CMakeError.log".
```

**Comment by akva2** [comment]
there were some cuda package updates. let me check..

**Comment by akva2** [comment]
jenkins build this please

**Comment by lisajulia** [comment]
Thanks everyone for input, reviews and merging :+1:

**Pull request merged**